libs/etcd_broker/src/lib.rs

```diff
@@ -55,7 +55,9 @@ pub struct SkTimelineInfo {
     #[serde(default)]
     pub peer_horizon_lsn: Option<Lsn>,
     #[serde(default)]
-    pub safekeeper_connection_string: Option<String>,
+    pub safekeeper_connstr: Option<String>,
+    #[serde(default)]
+    pub pageserver_connstr: Option<String>,
```

Comment:
I just follow the path of least resistance and don't change the code that is in safekeeper already.
Still not sure what should be better, @\kelvich proposed to have a "connected to compute" bit here instead, there might be better ideas on how to communicate pageserver's `NodeId` now that it has one.

Not sure we want to change it now, though?

Reply:
Pageserver constr is a new addition to etcd structure. So it is not already there from my point of view. 

Should we have a call then to discuss that? Because Option<String> allows only one pageserver, and the previous callmemaybe allowed to have more than one. 

Also do not get why it is a throwaway code

Reply:
> Pageserver constr is a new addition to etcd structure.

But an old one to the safekeeper's logic, where they set this connstr to a timeline data when it's ready for streaming.
I am sure I'll be able to change that logic too "the right way", but don't want to touch it now, if possible.

> Should we have a call then to discuss that?

We can, but, honestly, I'm sick of this PR already and the pressure on its merge timings, better help me with the last failing test first 🙂 
In general, I see this particular thing as an improvement to the existing code, it can work without it.
Tests are crucial and block everything, on the other hand.

> Because Option allows only one pageserver, and the previous callmemaybe allowed to have more than one.

Not sure this is right: the SK code logic implies that a given SK Timeline has the same `Option` in its data and it was propagated this way by the old safekeepers to pageservers via callmemaybe.
Now, same option gets pushed by multiple safekeeprs (with possible multiple values), so nothing had really changed: a single Pageserver is able to select from multiple etcd connstrs the same way it was able to select from multiple callmemaybe connstrs before.

> Also do not get why it is a throwaway code

Mainly, since now we don't need to construct such string at all: AFAIK, it's needed to identify connecting pageservers and to indicate that the SK timeline is ready to be WAL streamed.
I use the latter, but the former could/should be done better via `NodeId` of the pageserver passed via etcd communications (as one of the ways).

Reply:
Hmmm, I think you're right. I'm reading it wrong. This is equal to pageserver that compute is connected to, so this should be ok in our current paradigm when tenant relocation requires compute restart. So after restart compute is connected to different PS and this connstring should be changed accordingly. 

> I use the latter, but the former could/should be done better via NodeId

Yeah, this case was one of the reasons to add node id to use it instead of connection string to identify pageserver.

Thanks!